JSON logger doesn't print stacktrace anymore

### Describe the bug

Since commit https://github.com/quarkusio/quarkus/commit/697ea4ff090e83645d8a440aa639d7c2d0321c81 was merged the json logger doesn't print the stacktrace of out Exceptions.

This is the stacktrace:
```
LogManager error of type FORMAT_FAILURE: Formatting error

java.lang.RuntimeException: java.lang.NullPointerException
	at org.jboss.logmanager.formatters.StructuredFormatter.format(StructuredFormatter.java:277)
	at org.jboss.logmanager.ExtFormatter.format(ExtFormatter.java:58)
	at org.jboss.logmanager.handlers.WriterHandler.doPublish(WriterHandler.java:58)
	at org.jboss.logmanager.ExtHandler.publish(ExtHandler.java:88)
	at org.jboss.logmanager.ExtHandler.publishToNestedHandlers(ExtHandler.java:125)
	at io.quarkus.bootstrap.logging.QuarkusDelayedHandler.doPublish(QuarkusDelayedHandler.java:81)
	at org.jboss.logmanager.ExtHandler.publish(ExtHandler.java:88)
	at org.jboss.logmanager.LoggerNode.publish(LoggerNode.java:438)
	at org.jboss.logmanager.LoggerNode.publish(LoggerNode.java:480)
	at org.jboss.logmanager.LoggerNode.publish(LoggerNode.java:480)
	at org.jboss.logmanager.LoggerNode.publish(LoggerNode.java:480)
	at org.jboss.logmanager.LoggerNode.publish(LoggerNode.java:480)
	at org.jboss.logmanager.LoggerNode.publish(LoggerNode.java:480)
	at org.jboss.logmanager.LoggerNode.publish(LoggerNode.java:480)
	at org.jboss.logmanager.Logger.logRaw(Logger.java:921)
	at org.jboss.logmanager.Logger.log(Logger.java:884)
	at org.jboss.logging.JBossLogManagerLogger.doLogf(JBossLogManagerLogger.java:56)
	at org.jboss.logging.Logger.errorf(Logger.java:1731)
	at ***
	at ***
	at ***
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:73)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:62)
	at io.quarkus.arc.impl.ActivateRequestContextInterceptor.proceedWithStage(ActivateRequestContextInterceptor.java:61)
	at io.quarkus.arc.impl.ActivateRequestContextInterceptor.lambda$invokeStage$1(ActivateRequestContextInterceptor.java:44)
	at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1208)
	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2368)
	at java.base/java.util.concurrent.CompletableFuture$MinimalStage.thenCompose(CompletableFuture.java:2967)
	at io.quarkus.arc.impl.ActivateRequestContextInterceptor.invokeStage(ActivateRequestContextInterceptor.java:44)
	at io.quarkus.arc.impl.ActivateRequestContextInterceptor.aroundInvoke(ActivateRequestContextInterceptor.java:31)
	at io.quarkus.arc.impl.ActivateRequestContextInterceptor_Bean.intercept(Unknown Source)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:42)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:30)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:27)
	at ***
	at ***
	at ***
	at io.smallrye.reactive.messaging.providers.AbstractMediator.lambda$invokeBlocking$15(AbstractMediator.java:194)
	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromDeferredSupplier.subscribe(UniCreateFromDeferredSupplier.java:25)
	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
	at io.smallrye.mutiny.groups.UniSubscribe.withSubscriber(UniSubscribe.java:51)
	at io.smallrye.mutiny.groups.UniSubscribe.with(UniSubscribe.java:110)
	at io.smallrye.mutiny.groups.UniSubscribe.with(UniSubscribe.java:88)
	at io.vertx.mutiny.core.Context$1.handle(Context.java:172)
	at io.vertx.mutiny.core.Context$1.handle(Context.java:170)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$5(ContextImpl.java:205)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:270)
	at io.vertx.core.impl.ContextImpl$1.execute(ContextImpl.java:221)
	at io.vertx.core.impl.WorkerTask.run(WorkerTask.java:56)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:81)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$15.runWith(VertxCoreRecorder.java:695)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1474)

Caused by: java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:220)
	at java.base/java.util.ImmutableCollections$SetN.contains(ImmutableCollections.java:1183)
	at io.quarkus.logging.json.runtime.JsonFormatter$FormatterJsonGenerator.startObject(JsonFormatter.java:390)
	at org.jboss.logmanager.formatters.StructuredFormatter.addStackTraceElements(StructuredFormatter.java:534)
	at org.jboss.logmanager.formatters.StructuredFormatter.addException(StructuredFormatter.java:498)
	at org.jboss.logmanager.formatters.StructuredFormatter.format(StructuredFormatter.java:244)
	... 58 more
```


### Expected behavior

We expect to have the complete json logs as they where in previous versions.

### Actual behavior

The stacktraces are often not included in the logs.

### How to Reproduce?

Sorry for pasting the reproducer in the issue but the upload failed.

pom.xml
```
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-logging-json</artifactId>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-arc</artifactId>
        </dependency>
```

Reproducer.java
```
package ch.postfinance.orca;

import io.quarkus.logging.Log;
import io.quarkus.runtime.StartupEvent;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.enterprise.event.Observes;

@ApplicationScoped
public class Reproducer {
    void triggerBug(@Observes StartupEvent startupEvent) {
        try {
            throw new ReproducerException();
        } catch (ReproducerException e) {
            Log.warnf(e, "This is a reproducer (%s)", e.getMessage());
        }
    }

    public static class ReproducerException extends RuntimeException {
    }
}
```

Steps to reproduce:
1. mvn quarkus:dev
2. check the console logs

### Output of `uname -a` or `ver`

none

### Output of `java -version`

openjdk version "25.0.2" 2026-01-20 LTS OpenJDK Runtime Environment Temurin-25.0.2+10 (build 25.0.2+10-LTS) OpenJDK 64-Bit Server VM Temurin-25.0.2+10 (build 25.0.2+10-LTS, mixed mode, sharing)

### Quarkus version or git rev

3.36.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.16 (2bdd9fddda4b155ebf8000e807eb73fd829a51d5)

### Additional information

_No response_

--- a/extensions/logging-json/deployment/src/test/java/io/quarkus/logging/json/ConsoleJsonFormatterDefaultConfigTest.java
+++ b/extensions/logging-json/deployment/src/test/java/io/quarkus/logging/json/ConsoleJsonFormatterDefaultConfigTest.java
@@ -9,12 +9,16 @@ import java.util.logging.Formatter;
 import java.util.logging.Handler;
 import java.util.logging.Level;
 import java.util.logging.LogManager;
+import java.util.logging.LogRecord;
 import java.util.logging.Logger;
 
 import org.jboss.logmanager.formatters.StructuredFormatter;
 import org.jboss.logmanager.handlers.ConsoleHandler;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
 
 import io.quarkus.bootstrap.logging.InitialConfigurator;
 import io.quarkus.bootstrap.logging.QuarkusDelayedHandler;
@@ -39,6 +43,20 @@ public class ConsoleJsonFormatterDefaultConfigTest {
         assertThat(jsonFormatter.isPrintDetails()).isFalse();
         assertThat(jsonFormatter.getExcludedKeys()).isEmpty();
         assertThat(jsonFormatter.getAdditionalFields().entrySet()).isEmpty();
+    }
+
+    @Test
+    public void exceptionFormattingWorksWithDefaultConfig() throws Exception {
+        JsonFormatter jsonFormatter = getJsonFormatter();
+
+        LogRecord record = new LogRecord(Level.WARNING, "Something went wrong");
+        record.setThrown(new RuntimeException("boom"));
+
+        String line = jsonFormatter.format(record);
+
+        JsonNode node = new ObjectMapper().readTree(line);
+        assertThat(node.has("message")).isTrue();
+        assertThat(node.has("exception")).isTrue();
     }
 
     public static JsonFormatter getJsonFormatter() {

--- a/extensions/logging-json/deployment/src/test/java/io/quarkus/logging/json/ConsoleJsonFormatterJsonProviderExcludedKeyTest.java
+++ b/extensions/logging-json/deployment/src/test/java/io/quarkus/logging/json/ConsoleJsonFormatterJsonProviderExcludedKeyTest.java
@@ -88,6 +88,18 @@ public class ConsoleJsonFormatterJsonProviderExcludedKeyTest {
         assertThat(node.has("customfield")).isTrue();
     }
 
+    @Test
+    public void exceptionFormattingWorksWithExcludedKeys() throws Exception {
+        LogRecord record = new LogRecord(Level.WARNING, "Something went wrong");
+        record.setThrown(new RuntimeException("boom"));
+
+        String line = getJsonFormatter().format(record);
+
+        JsonNode node = new ObjectMapper().readTree(line);
+        assertThat(node.has("message")).isTrue();
+        assertThat(node.has("exception")).isTrue();
+    }
+
     @ApplicationScoped
     public static class NestedObjectProvider implements JsonProvider {
 

--- a/extensions/logging-json/runtime/src/main/java/io/quarkus/logging/json/runtime/JsonFormatter.java
+++ b/extensions/logging-json/runtime/src/main/java/io/quarkus/logging/json/runtime/JsonFormatter.java
@@ -281,7 +281,7 @@ public class JsonFormatter extends org.jboss.logmanager.formatters.JsonFormatter
         }
 
         public JsonLogGenerator startObject(final String key) throws Exception {
-            if (skippedDepth > 0 || excludedKeys.contains(key)) {
+            if (skippedDepth > 0 || (key != null && excludedKeys.contains(key))) {
                 skippedDepth++;
             } else {
                 delegate.startObject(key);
@@ -299,7 +299,7 @@ public class JsonFormatter extends org.jboss.logmanager.formatters.JsonFormatter
         }
 
         public JsonLogGenerator startArray(final String key) throws Exception {
-            if (skippedDepth > 0 || excludedKeys.contains(key)) {
+            if (skippedDepth > 0 || (key != null && excludedKeys.contains(key))) {
                 skippedDepth++;
             } else {
                 delegate.startArray(key);
@@ -407,7 +407,7 @@ public class JsonFormatter extends org.jboss.logmanager.formatters.JsonFormatter
 
         @Override
         public Generator startObject(final String key) throws Exception {
-            if (skippedDepth > 0 || excludedKeys.contains(key)) {
+            if (skippedDepth > 0 || (key != null && excludedKeys.contains(key))) {
                 skippedDepth++;
             } else {
                 delegate.startObject(key);
@@ -427,7 +427,7 @@ public class JsonFormatter extends org.jboss.logmanager.formatters.JsonFormatter
 
         @Override
         public Generator startArray(final String key) throws Exception {
-            if (skippedDepth > 0 || excludedKeys.contains(key)) {
+            if (skippedDepth > 0 || (key != null && excludedKeys.contains(key))) {
                 skippedDepth++;
             } else {
                 delegate.startArray(key);

